src/cascadia/TerminalApp/TabManagement.cpp

```diff
@@ -760,6 +760,58 @@ namespace winrt::TerminalApp::implementation
         }
     }
 
+    // Method Description:
+    // - Disables read-only mode on pane if the user wishes to close it and read-only mode is enabled.
+    // Arguments:
+    // - pane: the pane that is about to be closed.
+    // Return Value:
+    // - bool indicating whether the (read-only) pane can be closed.
+    winrt::Windows::Foundation::IAsyncOperation<bool> TerminalPage::_PaneConfirmCloseReadOnly(std::shared_ptr<Pane> pane)
+    {
+        if (const auto control{ pane->GetTerminalControl() })
+        {
+            if (control.ReadOnly())

```

Comment:
This used to be a `pane->ContainsReadOnly` call, replaced now with checking if the actual control in the pane is read-only. 

Technically, you can select a _parent_ pane. So you could close a parent pane (and close all the leaves within it). I think with this change, we've lost the ability to check if _any_ leaves in a pane are read-only. 

A simple fix might be to just add an `else (pane->ContainsReadOnly())` after the `if (const auto control` block, and repeat the walking we had before in that block. A little duplicated, but

Reply:
As the 'response' to a read-only pane is the same in this case (for parents and leaf nodes), I replaced the existing read-only check with the `pane->ContainsReadOnly` check and changed the toggling accordingly to avoid duplication (f5a1347)